Makefile

```diff
@@ -11,27 +6,25 @@ PREFIX ?= /usr/local
 MANPREFIX = $(PREFIX)/share/man
 
 # autoreload backend: inotify/nop
-AUTORELOAD = inotify
+AUTORELOAD ?= inotify
+
+# optional dependencies, see README for more info
+HAVE_LIBGIF ?= 1
+HAVE_LIBEXIF ?= 1
 
 # CFLAGS, any optimization flags goes here
 CFLAGS ?= -std=c99 -Wall -pedantic
 
-ifeq ($(HAVE_LIBEXIF), 1)
-	OPTIONAL_LIBS += -lexif
-else
-	HAVE_LIBEXIF = 0
-endif
-ifeq ($(HAVE_LIBGIF), 1)
-	OPTIONAL_LIBS += -lgif
-else
-	HAVE_LIBGIF = 0
-endif
-
 CPPFLAGS = -D_XOPEN_SOURCE=700 \
   -DHAVE_LIBGIF=$(HAVE_LIBGIF) -DHAVE_LIBEXIF=$(HAVE_LIBEXIF) \
   -I/usr/include/freetype2 -I$(PREFIX)/include/freetype2
 
-LDLIBS = -lImlib2 -lX11 -lXft -lfontconfig $(OPTIONAL_LIBS)
+lib_exif_0 =
+lib_exif_1 = -lexif
+lib_gif_0 =
+lib_gif_1 = -lgif
+LDLIBS = -lImlib2 -lX11 -lXft -lfontconfig \
+  $(lib_exif_$(HAVE_LIBEXIF)) $(lib_gif_$(HAVE_LIBGIF))
```

Comment:
Would be cool to have a way of specifying additional  libraries. For example, on FreeBSD it's necessary to append `-linotify` to the this list.

Reply:
We can use `+=` instead.

Anyways is there any other distro you can think of where this will be useful? I don't think we want to make changes if it's just *one* distro specific problem. If there are more than we're OK making changes.

Reply:
I'm not sure about the other but OpenBSD would benefit from this interface as well: http://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/graphics/sxiv/Makefile?rev=1.17&content-type=text/x-cvsweb-markup
-

Reply:
Homebrew is also modifying LDLIBS: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/sxiv.rb

Reply:
Excellent, thanks.